enhancements/run-bundle.md

```diff
@@ -0,0 +1,263 @@
+---
+title: run-bundle
+authors:
+  - "@jmrodri"
+reviewers:
+  - "@joelanford"
+  - "@estroz"
+  - "@kevinrizza"
+  - "@ecordell"
+  - "@shawn-hurley"
+approvers:
+  - "@joelanford"
+  - "@estroz"
+  - "@ecordell"
+  - "@shawn-hurley"
+creation-date: 2020-05-07
+last-updated: 2020-05-07
+status: implementable
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# Run Bundle
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+1. We currently check OLM installation status with `operator-sdk olm status`.
+   Can we change that to using the discovery API?
+   * Based on discussions, using the discovery API is the way to go.
+
+2. How do we handle --mode= flag on registry add
+   * should we ALWAYS default to --mode=semver?
+     * this has less testing and isnot yet the default. We should stick with
+        the default.
```

Comment:
That's not something that the registry or OLM itself supports. I recommend using semver mode if you want to be able to add arbitrary bundles like this.

Reply:
This should probably be configurable in the libraries being called (in fact, I think `permissive` will allow it). There's no technical reason to enforce it, other than that 9/10 times adding an operator that `replaces` something not in a catalog is a mistake.

Reply:
I should clarify - there are lots of valid reasons (especially for the pipeline) to not enforce the `replaces` field, because it allows you to truncate catalogs at a particular point in time and ship smaller diffs.

If we added support for skipping this validation in the opm libraries, the pipeline will have the option to disable it, and therefore sdk should have the option to disable it.

Reply:
So where do we end up on this? The questions I want us to answer are:
1. When adding a bundle to an empty index, what mode should be the default?
2. When adding a bundle to a provided index, what mode should be the default?
3. Should the add mode be configurable on the command line of `run bundle`?

Reply:
1. default is semver
2. replaces
3. flag to select that is optional not in an example for right now

Reply:
Based on all our discussions.
* When adding a bundle to an empty index, default to **semver**
* When adding a bundle to a provided index, default to **replaces**
* Should the add mode be configurable on the CLI? **optional flag**; not in an example right now